### ISSUE TITLE
yq e removes '---' header

### ISSUE THREAD

**Issue by KauzClay** [mikefarah/yq#1462]
**Describe the bug**
When I use yq, it removes the `---` marker of my yaml document

Version of yq: 4.30.5
Operating system: mac
Installed via: homebrew

**Input Yaml**

example.yaml:
```yaml
#! some ytt comments 
#@ a = b

---
kind: First
value: 1
---
kind: Second
value: 1

```

**Command**

```
❯ yq e  ' select(documentIndex == 0 ) ' example.yaml

```

**Actual behavior**

```yaml
#! some ytt comments
#@ a = b
kind: First
value: 1
```

**Expected behavior**

```yaml
#! some ytt comments
#@ a = b

---
kind: First
value: 1
```

**Additional context**
Sorry to bring this back up again, I'd gone through https://github.com/mikefarah/yq/issues/351 and https://github.com/mikefarah/yq/issues/1056 before reporting.

Strangely, if I don't have that empty line before the `---` and input document looks like this:
```yaml
#! some ytt comments 
#@ a = b
---
kind: First
value: 1
---
kind: Second
value: 1

```

Then it behaves how I'd expect:
```
❯ yq e  ' select(documentIndex == 0 ) ' /tmp/example.yaml
#! some ytt comments
#@ a = b
---
kind: First
value: 1
```

**Comment by mikefarah** [comment]
Found the bug, will fix in the next release

**Comment by mikefarah** [comment]
Fixed in v4.30.6
